File: app/src/main/java/test/notificationreader/model/notifications/NotificationFactory.java
package test.notificationreader.model.notifications;

public class NotificationFactory {

    public static Notification build(String text, String aPackage) {
        if (text.contains(":")) {
            return new MessageNotification(text, aPackage);
        } else if (aPackage.contains("android")) {
            return new SystemNotification(text, aPackage);
        }

        return new Notification(text, aPackage);
    }

}

File: app/src/main/java/test/notificationreader/model/notifications/NotificationFactory.kt
package test.notificationreader.model.notifications

object NotificationFactory {
    fun build(text: String, aPackage: String): Notification {
        if (text.contains(":")) {
            return MessageNotification(text, aPackage)
        } else if (aPackage.contains("android")) {
            return SystemNotification(text, aPackage)
        }

        return Notification(text, aPackage)
    }
}
